src/marten/handlers/schema.cr

```diff
@@ -75,7 +78,19 @@ module Marten
       end
 
       def post
-        schema.valid? ? process_valid_schema : process_invalid_schema
+        @response = run_before_validation_callbacks
+
```

Comment:
Here I think we could return early and avoid calling `Schema#valid?` if we already have a response. For example:

```suggestion
        @response = run_before_validation_callbacks
        return response! if !@response.nil?

```